jaeger-core/src/test/java/io/jaegertracing/internal/JaegerSubclassTest.java

```diff
@@ -0,0 +1,231 @@
+                                          Sampler sampler,
+                                          PropagationRegistry registry,
+                                          Clock clock,
+                                          Metrics metrics,
+                                          Map<String, Object> tags,
+                                          boolean zipkinSharedRpcSpan,
+                                          ScopeManager scopeManager,
+                                          BaggageRestrictionManager baggageRestrictionManager,
+                                          boolean expandExceptionLogs,
+                                          JaegerObjectFactory objectFactory) {
+        return new CustomTracer(
+            serviceName,
+            reporter,
+            sampler,
+            registry,
+            clock,
+            metrics,
+            tags,
+            zipkinSharedRpcSpan,
+            scopeManager,
+            baggageRestrictionManager,
+            expandExceptionLogs,
+            objectFactory);
+      }
+    }
+
+    public class CustomSpanBuilder extends JaegerTracer.SpanBuilder {
+      protected CustomSpanBuilder(String operationName) {
+        super(operationName);
+      }
+
+      @Override
+      public CustomSpan start() {
+        return (CustomSpan) super.start();
+      }
+    }
+
+    public CustomTracer(
+        String serviceName,
+        Reporter reporter,
+        Sampler sampler,
+        PropagationRegistry registry,
+        Clock clock,
+        Metrics metrics,
+        Map<String, Object> tags,
+        boolean zipkinSharedRpcSpan,
+        ScopeManager scopeManager,
+        BaggageRestrictionManager baggageRestrictionManager,
+        boolean expandExceptionLogs,
+        JaegerObjectFactory objectFactory) {
+      super(
+          serviceName,
+          reporter,
+          sampler,
+          registry,
+          clock,
+          metrics,
+          tags,
+          zipkinSharedRpcSpan,
+          scopeManager,
+          baggageRestrictionManager,
+          expandExceptionLogs,
+          objectFactory);
+    }
+  }
+
+  private static class CustomSpan extends JaegerSpan {
+    public CustomSpan(
+        JaegerTracer tracer,
+        String operationName,
+        JaegerSpanContext context,
+        long startTimeMicroseconds,
+        long startTimeNanoTicks,
+        boolean computeDurationViaNanoTicks,
+        Map<String, Object> tags,
+        List<Reference> references) {
+      super(
+          tracer,
+          operationName,
+          context,
+          startTimeMicroseconds,
+          startTimeNanoTicks,
+          computeDurationViaNanoTicks,
+          tags,
+          references);
+    }
+
+    @Override
+    public CustomSpanContext context() {
+      return (CustomSpanContext) super.context();
+    }
+  }
+
+  private static class CustomSpanContext extends JaegerSpanContext {
+    public CustomSpanContext(
```

Comment:
could it be private? go for minimum visibility

Reply:
I think so. Didn't want to change visibility from base class. Constructors should be okay with that.